src/Raven.Server/Documents/Indexes/MapIndexBase.cs

```diff
@@ -53,9 +54,16 @@ public override int HandleMap(LazyStringValue lowerId, LazyStringValue id, IEnum
             EnsureValidStats(stats);
 
             bool mustDelete;
-            using (_stats.BloomStats.Start())
+            if (_filters.Consumed == false)
             {
-                mustDelete = _filters.Add(lowerId) == false;
+                using (_stats.BloomStats.Start())
+                {
+                    mustDelete = _filters.Add(lowerId) == false;
```

Comment:
I would rather prefer to handle everything inside the filters instead of adding this logging outside